src/classes/CRLP_RollupQueueable.cls

```diff
@@ -92,20 +92,31 @@ public class CRLP_RollupQueueable implements System.Queueable {
      */
     public static void runRollupsForIds(List<Id> summaryRecordIds) {
         SObjectType objType = summaryRecordIds[0].getSobjectType();
+        List<CRLP_RollupProcessingOptions.RollupType> jobTypesToRun = new List<CRLP_RollupProcessingOptions.RollupType>();
 
         CRLP_RollupProcessingOptions.ProcessingOptions options = new CRLP_RollupProcessingOptions.ProcessingOptions();
         options.mode = CRLP_RollupProcessingOptions.BatchJobMode.NonSkewMode;
         options.parentObjType = objType;
         options.detailObjType = Opportunity.SObjectType;
         if (objType == Account.SObjectType) {
-            options.parentKeyField = SObjectType.Opportunity.fields.AccountId.getName();
-            options.rollupJobType = CRLP_RollupProcessingOptions.RollupType.AccountHardCredit;
+            jobTypesToRun.add(CRLP_RollupProcessingOptions.RollupType.AccountHardCredit);
+            if (CRLP_Rollup_SVC.hasActiveAccountSoftCreditRollupsForSObject(objType)){
+                jobTypesToRun.add(CRLP_RollupProcessingOptions.RollupType.AccountSoftCredit);
+            }
+            if (CRLP_Rollup_SVC.hasActivePartialSoftCreditRollupsForSObject(objType)) {
+                jobTypesToRun.add(CRLP_RollupProcessingOptions.RollupType.AccountContactSoftCredit);
+            }
+            //options.rollupJobType = CRLP_RollupProcessingOptions.RollupType.AccountHardCredit;
         } else if (objType == Contact.SObjectType) {
             options.parentKeyField = SObjectType.Opportunity.fields.Primary_Contact__c.getName();
-            options.rollupJobType = CRLP_RollupProcessingOptions.RollupType.ContactHardCredit;
+            jobTypesToRun.add(CRLP_RollupProcessingOptions.RollupType.ContactHardCredit);
+            if (CRLP_Rollup_SVC.hasActivePartialSoftCreditRollupsForSObject(objType)) {
```

Comment:
We previously agreed not to roll up soft credits on the Contact via a trigger. I did some experiments with this early on and discovered that you cannot run the hard credit and soft credit rollups at the same time - which is what this code is trying to do.  We pulled out the original code I wrote to do this once I put this into action in one of the larger orgs. You'll need to pull out this code from the AccountSoftCredit logic as well since it'll have exactly the same conflict with AccountHardCredit.

Reply:
@bethbrains  The trigger rollup logic still needs to be pulled out.